frontend/src/components/dashboard/create/email/tests/EmailTemplate.test.tsx

```diff
@@ -0,0 +1,259 @@
+import React from 'react'
+import {
+  Campaign,
+  screen,
+  mockCommonApis,
+  server,
+  render,
+  fireEvent,
+} from 'test-utils'
+import CampaignContextProvider from 'contexts/campaign.context'
+import FinishLaterModalContextProvider from 'contexts/finish-later.modal.context'
+import userEvent from '@testing-library/user-event'
+import { Route } from 'react-router-dom'
+import EmailTemplate from '../EmailTemplate'
+
+function mockApis(protect: boolean) {
+  const campaign: Campaign = {
+    id: 1,
+    name: 'Test email campaign',
+    type: 'EMAIL',
+    created_at: new Date(),
+    valid: false,
+    protect,
+    demo_message_limit: null,
+    csv_filename: null,
+    is_csv_processing: false,
+    num_recipients: null,
+    job_queue: [],
+    halted: false,
+    has_credential: false,
+  }
+  const { handlers } = mockCommonApis({
+    // Start with a freshly created email campaign
+    campaigns: [campaign],
+  })
+  return handlers
+}
+
+function renderTemplatePage() {
+  const setActiveStep = jest.fn()
+
+  render(
+    <Route path="/campaigns/:id">
+      <CampaignContextProvider>
+        <FinishLaterModalContextProvider>
+          <EmailTemplate setActiveStep={setActiveStep} />
+        </FinishLaterModalContextProvider>
+      </CampaignContextProvider>
+    </Route>,
+    {
+      router: { initialIndex: 0, initialEntries: ['/campaigns/1'] },
+    }
+  )
+}
+
+test('displays the necessary elements', async () => {
+  // Setup
+  server.use(...mockApis(false))
+  renderTemplatePage()
+
+  // Wait for the component to fully load
+  const heading = await screen.findByRole('heading', {
+    name: /create email message/i,
+  })
+
+  /**
+   * Assert that the following elements are present:
+   * 1. "Create email message" heading
+   * 2. From address dropdown
+   * 3. Subject textbox
+   * 4. Message textbox
+   * 5. Reply-to textbox
+   * 6. Next button
+   */
+  expect(heading).toBeInTheDocument()
+  expect(
+    screen.getByRole('textbox', {
+      name: /subject/i,
+    })
+  ).toBeInTheDocument()
+  expect(
+    screen.getByRole('textbox', { name: /rdw-editor/i })
+  ).toBeInTheDocument()
+  expect(
+    screen.getByRole('textbox', {
+      name: /replies/i,
+    })
+  )
+  expect(screen.getByRole('button', { name: /next/i })).toBeInTheDocument()
+})
+
+test('displays an error if the subject is empty after sanitization', async () => {
+  // Setup
+  jest.spyOn(console, 'error').mockImplementation(() => {
+    // Do nothing. Mock console.error to silence expected errors
+    // due to submitting invalid templates to the API
+  })
+  server.use(...mockApis(false))
+  renderTemplatePage()
```

Comment:
It's a bit difficult to do so, considering how each test has slightly different setup and teardown processes.

Reply:
I think it's a bit difficult to do so, since most tests have slightly different setup/teardown phases. For example, some tests mock `console.error` and some use protected email campaigns.

Reply:
Alright, as for mocking the `console.error`, if you want to do this globally, you can feed it in a setup file in your `jest.config.ts`, this is something the backend tests currently implement to silence `console.log` messages

Reply:
Good point. I considered doing so, but decided against mocking `console.error` globally. When fixing test failures, I found that `console.error` statements provided useful information.

Reply:
Sure, the backend tests currently implement that only for `console.log` to not overcrowd the travis log output